### PR TITLE
Symfony 7 support

### PR DESCRIPTION
**Pull request by l-you**
Fixes https://github.com/FriendsOfSymfony/FOSElasticaBundle/issues/1932. Add PHP 8.3 tests.

This PR replaces https://github.com/FriendsOfSymfony/FOSElasticaBundle/pull/1931 and https://github.com/FriendsOfSymfony/FOSElasticaBundle/pull/1930.

### CODE REVIEW

**Comment by rusted-love** [comment]
There is an issue with dependencies.
`doctrine/phpcr-odm` locked to `symfony/console@6.0`. 
There is a ready PR that adds support for Symfony 7 in `doctrine/phpcr-odm`. Take a look at https://github.com/doctrine/phpcr-odm/pull/846.

**Comment by rusted-love** [comment]
The second reason tests are failing is that `jms/serializer-bundle` has no support for Symfony 7 components.
There is a draft PR that would fix that: https://github.com/schmittjoh/JMSSerializerBundle/pull/937 but merge state is unknown.

**Comment by rusted-love** [comment]
Current PR will be updated after the following PR is merged.
- https://github.com/schmittjoh/JMSSerializerBundle/pull/937
- https://github.com/doctrine/phpcr-odm/pull/846

File: .github/workflows/continuous-integration.yaml
Comment (evertharmeling) [inline .github/workflows/continuous-integration.yaml:47]: All references to Symfony `6.2` may be updated to `6.3` as `6.2` is unmaintained [since july 2023](https://symfony.com/releases/6.2)

File: composer.json
Comment (evertharmeling) [inline composer.json:38]: Update all Symfony references to `^6.3` see previous comment

File: composer.json
Comment (evertharmeling) [inline composer.json:37]: Add `|| ^7.0`

**Comment by evertharmeling** [comment]
@XWB do you or somebody else be able to check this PR and move it forward? @rusted-love could you check the Code style errors in the CI flow (/ re-push code to re-run the CI which should have the other errors fixed since the mentioned blocked issues are merged)? 🚀

**Comment by XWB** [comment]
I can only merge when all pipelines have passed.

**Comment by rusted-love** [comment]
> @rusted-love could you check the Code style errors in the CI flow (/ re-push code to re-run the CI which should have the other errors fixed since the mentioned blocked issues are merged)? 🚀 

@evertharmeling Yes, https://github.com/doctrine/phpcr-odm/pull/846 was merged a few days ago, but it was not released under next minor version 1.8.0 -> 1.8.1. Still waiting for a release tag.

File: .gitignore
Comment (Seb33300) [inline .gitignore:9]: Please remove this.
It should be added to your global .gitignore

File: src/Index/AliasProcessor.php
Comment (Seb33300) [inline src/Index/AliasProcessor.php:183]: Space is missing?

**Review by Seb33300 — CHANGES_REQUESTED** [review]

File: .gitignore
Comment (rusted-love) [reply .gitignore:9]: Okay, I'll do it later today.

**Comment by Seb33300** [comment]
What version of PHP-CS-fixer did you use to update the code style?
The workflow is using version 3.8.0
https://github.com/FriendsOfSymfony/FOSElasticaBundle/blob/f44abba6cf6a902ddcbbaac5acc2bdac7d8b318d/.github/workflows/continuous-integration.yaml#L17
You may need to update the workflow version to prevent it to fail

**Comment by llupa** [comment]
> but I need a community suggestion how to move forward. 
  
I think first it would be great if you can run php-cs-fixer at `3.8.0` on your code and update this branch so `Continuous integration / Coding style (pull_request)` passes.

And **dbu** asked (in the same PR your linked): 

> have you been able to build FOSElasticaBundle with the latest dev-2.x version? that would be a good check to know if we break anything for you.

Since you have all this work done, you could switch to their upcoming v2 and see if this pipeline will keep working. I personally would update in this same PR and mark it as Draft just in case. Or if you feel unsure, make another branch out of this branch and try it out.

If the tests pass, pinging David will only add reassurance for his ~planned release end of this month. 

What do you think? @rusted-love

**Comment by rusted-love** [comment]
@llupa I've done what you said. There are issues with dependencies again. For some reason composer does not use RC-1 and beta releases. I would be glad to get help again. :)

**Comment by evertharmeling** [comment]
> For some reason composer does not use RC-1 and beta releases. I would be glad to get help again. :)

Probably due to not having `minimum-stablity` defined in the composer.json, which defaults to `stable` ([see](https://getcomposer.org/doc/04-schema.md#minimum-stability)).

```json
...
    "minimum-stability": "RC",
    "prefer-stable": true,
...
```

This should do the trick, allowing specific `RC` versions when defined, but rely on `stable` otherwise. But we should also be careful using `RC`/`beta` releases from a bundle perspective of course.

**Comment by XWB** [comment]
Closing in favor of https://github.com/FriendsOfSymfony/FOSElasticaBundle/pull/1938

**Pull request closed without merging**
